cloud/amazon/ec2_vpc_peering.py

```diff
@@ -0,0 +1,596 @@
+    """
+    Return a peering connection as dictionary.
+    """
+    if not connection:
+        return dict()
+
+    return {
+        'id': connection.id,
+        'expiration': connection.expiration_time or '',
+        'source_vpc_id':  connection.requester_vpc_info.vpc_id,
+        'source_vpc_cidr': connection.requester_vpc_info.cidr_block,
+        'peer_vpc_id': connection.accepter_vpc_info.vpc_id,
+        'peer_vpc_cidr': connection.accepter_vpc_info.cidr_block,
+        'status': connection.status_code,
+        'status_message': connection.status_message,
+        'tags': connection.tags
+    }
+
+
+def results_list_has_changed(results):
+    """
+    Given a list of results where each result is a tuple of the form
+    (changed, ...), [where changed is a boolean], return True if this group
+    has changed.
+    """
+    get_changed = itemgetter(0)
+    return any(filter(get_changed, results))
+
+
+def wait_until_condition(condition_check, interval=1):
+    """
+    Given a condition check as callable, sleep until the condition check
+    returns True
+    """
+    while True:
+        time.sleep(interval)
+        if condition_check():
+            break
+
+def main():
+
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+            region = dict(required=True),
+            name = dict(required=False),
+            source_vpc = dict(required=False),
+            peer_vpc = dict(required=False, aliases=['dest_vpc']),
+            peer_owner_id = dict(required=False),
+            state = dict(
+                choices=['absent', 'present', 'active', 'pending', 'list'],
+                default='present',
+                required=False),
+            force_accept = dict(required=False, type='bool', default=False),
+            ignore_rejection = dict(
+                required=False, type='bool', default=False),
+            always_delete = dict(required=False, type='bool', default=False)
+        )
+    )
+
+    module = AnsibleModule(argument_spec=argument_spec,
+        supports_check_mode=True)
+
+    region, ec2_url, boto_params = get_aws_connection_info(module)
+    desired_state = module.params.get('state')
+
+    vpcpmod = VPCPeeringModule(module, boto_params) 
+
+    # source_vpc and peer_vpc are only not required when state is 'list'
+    if (desired_state != 'list') and \
+            ((not module.params.get('source_vpc')) or \
+             (not module.params.get('peer_vpc'))):
+        module.fail_json(
+            msg='source_vpc and dest_vpc required unless state="list"')
+
+    if desired_state in ('list',):
+        action = lambda: vpcpmod.list_all()
+    elif desired_state in ('present', 'active'):
+        action = lambda: vpcpmod.ensure_present(want_active=True)
+    elif desired_state in ('pending',):
+        action = lambda: vpcpmod.ensure_present()
+    elif desired_state in ('absent',):
+        action = lambda: vpcpmod.ensure_absent()
+
+    try:
```

Comment:
I have pushed code with my fix here: https://github.com/ansible/ansible-modules-core/compare/devel...tendant:ec2_vpc_peering

You can find the branch here:

https://github.com/tendant/ansible-modules-core/tree/ec2_vpc_peering

The "fact module" is not changed. Since I am not sure about what is the best way to handle it. 

Let me know whether the code is good or not. I can create pull request if it is good to go.